bazel/external_proto_deps.bzl

```diff
@@ -29,10 +29,14 @@ EXTERNAL_PROTO_GO_BAZEL_DEP_MAP = {
     #    go_googleapis in https://github.com/bazelbuild/rules_go/blob/master/go/dependencies.rst#overriding-dependencies
     "@com_google_googleapis//google/api/expr/v1alpha1:checked_proto": "@go_googleapis//google/api/expr/v1alpha1:expr_go_proto",
     "@com_google_googleapis//google/api/expr/v1alpha1:syntax_proto": "@go_googleapis//google/api/expr/v1alpha1:expr_go_proto",
+    "@dev_cel//proto/cel/expr:checked_proto": "@dev_cel//proto/cel/expr:checked_go_proto",
```

Comment:
IIUC these are introduced in addition to the com_googleapis variant to avoid introducing a breaking API change. However, as this is a superset of the com_googleapis project, then I think that conceptually it needs to replace it completely (to avoid violating ODR).
I'm a bit concerned what will happen if these two repos diverge. but as we don't have versioning in place, I don't think we can do much about it.
cc @htuch if there are other ways here.

Reply:
@adisuissa Canonical CEL (cel.dev) is meant to replace `v1alpha1` CEL from google APIs package. Canonical CEL will always be extended in a backward-compatible way (as protos normally should be), so there isn't a divergence problem. I'm CCing CEL folks here for more context: @l46kok, @TristonianJones.
Also, @tyxia has more context from the Envoy side.

Reply:
From Envoy perspective, `cel.dev` and `v1alpha1` will co-exist. They are guaranteed to be synced by CEL team.

New users in Envoy are recommended to use `cel.dev` while existing users can still use `v1alpha1`

Thus, `v1alpha1` is deprecated but can not be removed.

Reply:
I think this makes sense. When the Envoy PR lands, which shoudl be adding implementation level support (not just API import), it might make sense to audit any remaining references to CEL that goes direct to `google.api` package namespace in the API protos in `envoyproxy/envoy`, e.g. RBAC.

Reply:
Looks like nothing is blocking this PR from being merged, right?